runtime/compiler/aarch64/codegen/CallSnippet.cpp

```diff
@@ -0,0 +1,590 @@
+               {
+               count++;
+               }
+            floatArgNum++;
+            break;
+         }
+      }
+
+   return count;
+   }
+
+static uint32_t encodeHelperBranchAndLink(TR::SymbolReference *symRef, uint8_t *cursor, TR::Node *node, TR::CodeGenerator *cg)
+   {
+   intptrj_t target = (intptrj_t)symRef->getMethodAddress();
+
+   if (cg->directCallRequiresTrampoline(target, (intptrj_t)cursor))
+      {
+      target = TR::CodeCacheManager::instance()->findHelperTrampoline(symRef->getReferenceNumber(), (void *)cursor);
+
+      TR_ASSERT_FATAL(TR::Compiler->target.cpu.isTargetWithinUnconditionalBranchImmediateRange(target, (intptrj_t)cursor),
+                      "Target address is out of range");
+      }
+
+   cg->addExternalRelocation(new (cg->trHeapMemory()) TR::ExternalRelocation(
+                             cursor,
+                             (uint8_t *)symRef,
+                             TR_HelperAddress, cg),
+                             __FILE__, __LINE__, node);
+
+   intptr_t distance = target - (intptr_t)cursor;
+   return TR::InstOpCode::getOpCodeBinaryEncoding(TR::InstOpCode::bl) | ((distance >> 2) & 0x3ffffff); /* imm26 */
+   }
+
+TR_RuntimeHelper TR::ARM64CallSnippet::getHelper()
+   {
+   TR::Compilation * comp = cg()->comp();
+   TR_J9VMBase *fej9 = (TR_J9VMBase *) (cg()->fe());
+   TR::Node *callNode = getNode();
+   TR::SymbolReference *methodSymRef = callNode->getSymbolReference();
+   TR::MethodSymbol    *methodSymbol = methodSymRef->getSymbol()->castToMethodSymbol();
+   TR::SymbolReference *glueRef = NULL;
+
+   if (methodSymRef->isUnresolved() || comp->compileRelocatableCode())
+      {
+      if (methodSymbol->isSpecial())
+         return TR_ARM64interpreterUnresolvedSpecialGlue;
+      if (methodSymbol->isStatic())
+         return TR_ARM64interpreterUnresolvedStaticGlue;
+      return TR_ARM64interpreterUnresolvedDirectVirtualGlue;
+      }
+
+   if (methodSymbol->isVMInternalNative() || methodSymbol->isJITInternalNative())
+      return TR_ARM64nativeStaticHelper;
+
+   bool synchronised = methodSymbol->isSynchronised();
+
+   TR::DataType dataType = callNode->getDataType();
+   switch (dataType)
+      {
+      case TR::NoType:
+         return synchronised ? TR_ARM64interpreterSyncVoidStaticGlue : TR_ARM64interpreterVoidStaticGlue;
+      case TR::Int32:
+         return synchronised ? TR_ARM64interpreterSyncIntStaticGlue : TR_ARM64interpreterIntStaticGlue;
+      case TR::Int64:
+      case TR::Address:
+         return synchronised ? TR_ARM64interpreterSyncLongStaticGlue : TR_ARM64interpreterLongStaticGlue;
+      case TR::Float:
+         return synchronised ? TR_ARM64interpreterSyncFloatStaticGlue : TR_ARM64interpreterFloatStaticGlue;
+      case TR::Double:
+         return synchronised ? TR_ARM64interpreterSyncDoubleStaticGlue : TR_ARM64interpreterDoubleStaticGlue;
+      default:
+         TR_ASSERT(false, "Bad return data type '%s' for a call node.\n",
```

Comment:
A fatal assert is probably warranted here.

Reply:
Changed to `TR_ASSERT_FATAL()`.